src/containers/editor/graph/Node.tsx

```diff
@@ -90,10 +90,10 @@ const KV = memo(({ index, property, valueClassName, valueText, hasChildren, widt
 
   return (
     <div className="graph-kv" style={{ width }}>
-      <div contentEditable="true" suppressContentEditableWarning className={cn("graph-k", keyClass)}>
+      <div className={cn("graph-k", keyClass)} title={keyText}>
```

Comment:
We should have a tooltip first, then use it to display values, rather than using title attribute, since it can't 
 have custom styles.

Reply:
@loggerhead Hello, my approach is to use context api, and use the component, ViewportPortal, to deal with zoom-in and zoom-out. Do you see any downside of this? If you have no objection, I will keep working on this to improve UX and UI.

Reply:
@haecheonlee I made a mistake; it should be a popover, not a tooltip.

Reply:
@haecheonlee There already have zoom-in and zoom-out in the bottom-left button group, what do you mean to deal with zoom-in and zoom-out?

Reply:
@loggerhead We can use the click event, and the size of a raw element is static regardless of zooming-in/zooming-out. However, by using the component, we avoid the need to manually calculate the size of the rendered element, and keep the relative size. 

Here's the link to the description of the component: [ViewportPortal](https://reactflow.dev/api-reference/components/viewport-portal)

Reply:
> @loggerhead Hello, my approach is to use context api, and use the component, ViewportPortal, to deal with zoom-in and zoom-out. Do you see any downside of this? If you have no objection, I will keep working on this to improve UX and UI.

@haecheonlee I'm okay as long as the performance doesn't decline.

Reply:
Alright, let me finish and i will let you review. Thank you

Reply:
@loggerhead Hello, can you review it?

Reply:
@haecheonlee I think there have several problems:

1. The text inside the popover couldn't be selected. We should use hover instead of click to trigger the popover. Clicking on the key or value will enter the edit status, so using click as the trigger will cause a conflict if we need support edit in the feature.
2. The popover always displays below, but I prefer it to default to display above. And It's better to adjust the position based on the window height to decide whether it should display above or below.
3. You need to adjust the popover overflow to prevent excessively long string values making height to become very large.
4. We currently need to preserve the `contenteditable` attribute. Since I added a search for graph mode yesterday, it will enter edit status to notify the user of the search position. I'll change it later to use a border as the notification method.
5. It's preferable to keep the color of the text in the popover the same as the graph node.

By the way, your work is a great begging for us to provide a preview feature in the future!

Reply:
@loggerhead Hello, I updated some of what you mentioned here

Update:
1. The onClick event is replaced by onMouseEnter and onMouseLeave
3. This now has the max-height set to 250px
4. The attributes are reverted
5. The class names of key and value now are passed to the state and will be set in the popover

Comment:
2. I am not sure what you mean by window height. And, putting the popover above each key/value will require extra calculation of the current popover's height to put it in the right position.

Reply:
@haecheonlee There still have several problems:
1. Unable to maintain popover display when moving the mouse inside it, so unable to select text within it. It's inconvenient for people who want to copy the value in the graph node.
2. We need to add a delay timeout for displaying the popover. It would be annoying if people accidentally trigger the popover by just moving the mouse over the graph node without intending to.

I mean you should show the popover above when near the bottom border of the graph, like this:

![image](https://github.com/user-attachments/assets/62b475f4-3cfd-4871-9d98-955d0235b163)

Also, you should show the popover bottom when near the top border.

Reply:
@loggerhead Hello, sorry I couldn't make time to work on this last week. Can you check the feature while I work on adding a test? 

Thank you,

Reply:
> this seems to have error with position... 

@haecheonlee You mean table view? By the way, I have added lots of tests in `e2e` branch, and I will merge it tomorrow. Maybe you can add tests based on it.

Reply:
You may find some e2e tests fail (I'm still working on it), it's okay if it's not related to your changes.

Reply:
@loggerhead Hello, I added the tests. You can check the tests I added.

However, I’m uncertain if I can implement the suggestion you made about allowing users to hover over the popover to copy/paste, as the ViewportPortal component itself seems to cause issues with other views. I am trying to make it work around with css and js events... but no luck yet. Let me try to figure it out, and if you have better suggestion I would appreciate it!

Reply:
@haecheonlee You could check `src/containers/editor/table/Tooltip.tsx`, I think the implement could be similar.

Reply:
@loggerhead Hello, it's so much better to use the existing components... but it seems that the test doesn't seem to work. I will update the test, meanwhile you can check the new implementation. Thank you.

Reply:
@loggerhead Hello, it's been a while to resolve it... can you review it? Thank you.

Reply:
@haecheonlee Thank you and I merged. I'm busy for debugging these days, sorry for slow reply.